### PR TITLE
Landing Entropy 0.2 (first part)

### PR DESCRIPTION
**Pull request by Yoric**
Ok, let's start landing entropy 0.2!

### CODE REVIEW

File: crates/binjs_io/src/lib.rs [hunk 1]
```diff
@@ -18,10 +18,13 @@ extern crate range_encoding;
 #[macro_use]
 extern crate serde_derive;
 extern crate serde;
+extern crate smallvec;
```
Comment (arai-a) [inline crates/binjs_io/src/lib.rs]: where is this used?

File: crates/binjs_io/src/entropy/write.rs [hunk 2]
```diff
@@ -9,8 +9,10 @@
 
 use ::TokenWriterError;
 use ::io::{ Path, TokenWriter };
-use ::io::statistics::{ ContentInfo, Instances };
+use ::io::statistics::{ Bytes, BytesAndInstances, ContentInfo, Instances };
 use bytes::lengthwriter::LengthWriter;
+use bytes::varnum::WriteVarNum;
+use super::probabilities::IntoStatistics;
```
Comment (arai-a) [inline crates/binjs_io/src/entropy/write.rs]: what are above 3 lines change for?

File: crates/binjs_io/src/bytes/float.rs [hunk 3]
```diff
@@ -90,6 +102,56 @@ impl<T> WriteVarFloat for T where T: Write {
...
+                    return Err(std::io::Error::new(std::io::ErrorKind::InvalidData, "Invalid varfloat"));
+                }
+            }
+        }
+        // This float value maps to a i32.
+        let as_i32 =
+            if as_u32 % 2 == 0 {
+                (as_u32 / 2) as i32
+            } else {
+                - (((as_u32 - 1) / 2) as i32)
```
[the remark this answers is not in the document]
Comment (Yoric) [reply crates/binjs_io/src/bytes/float.rs]: I wanted to make sure that numbers between -100 and +100 (which I'm pretty sure are the most common) get encoded in one single byte.

File: crates/binjs_io/src/lib.rs (hunk 1 above)
Comment (Yoric) [reply crates/binjs_io/src/lib.rs]: Might be in a followup patch.

File: crates/binjs_io/src/bytes/float.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (arai-a) [reply crates/binjs_io/src/bytes/float.rs]: sounds reasonable, but then please add comment to somewhere about the encoding details and the reasons.

**Pull request merged**